torch/_higher_order_ops/auto_functionalize.py

```diff
@@ -365,7 +375,20 @@ def __call__(
 auto_functionalized_v2.fallthrough(DispatchKey.AutogradCUDA)
 
 
-def can_auto_functionalize(op: OperatorBase) -> bool:
+def can_auto_functionalize(op: Union[OperatorBase, HigherOrderOperator]) -> bool:
+    if isinstance(op, HigherOrderOperator):
+        # HOPs that implements gen_shema is auto_functionalizable.
+        # However, it's up to each hop's functioanlization implementation to decide
+        # whether do_auto_functionalized is called when input mutation happens.
```

Comment:
Uh what does this mean?

Reply:
I just want to clarify that do_auto_functionalized is called in hop's functionalization key's implementation.

Reply:
see the  base_hop functionalization key: https://github.com/pytorch/pytorch/pull/151067/files#diff-193a1ca345897131c5b7bde79857e6c7c6a6b407284b82f6c680c3b3bd59d477R141-R144.  Sorry i mean it should be do_auto_functionalized_v2 not do_auto_functionalized, is it the confusion here?